### PR TITLE
Io 1043 Accordion component

### PR DESCRIPTION
**Pull request by cristain7**

### CODE REVIEW

File: src/components/accordion/accordion.scss [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  &__heading {
+    @extend %ods-text--size-hotel, %ods-text--size-golf-breakpoint-medium;
+    @extend %ods-text--weight-medium;
+  }
+
+  &__trigger {
+    @extend %ods-padding-5, %ods-padding-vertical-6-breakpoint-medium, %ods-padding-horizontal-8-breakpoint-medium;
+
+    border: none;
+    background-color: white;
```
Comment (kbergha) [inline src/components/accordion/accordion.scss:26]: Ikke `colors.$white`? (gjelder også linje 53 + 59 + 76 + 109)

File: src/general/helpers/collapsible/collapsible.init.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import OsgCollapsible from './collapsible';
```
Comment (kbergha) [inline src/general/helpers/collapsible/collapsible.init.js:1]: Osg? Er det ikke Ods nå (også linje 4)?

File: src/components/accordion/accordion.scss (hunk 1 above)
Comment (cristain7) [reply src/components/accordion/accordion.scss]: Oppdatert!

File: src/general/helpers/collapsible/collapsible.init.js (hunk 2 above)
Comment (cristain7) [reply src/general/helpers/collapsible/collapsible.init.js]: Oppdatert det!

**Review by kbergha — COMMENTED** [review]
Siden dette er et nytt komponent, er det testet på flere enheter og nettlesere for rariteter?

File: src/general/helpers/collapsible/collapsible.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<div class="ods-devtools-nondevmode">
```
Comment (larsboldt) [inline src/general/helpers/collapsible/collapsible.html:1]: There should be examples of all options in developer tab so its easy to see that the code is working

File: src/general/helpers/collapsible/collapsible.js [hunk 4]
```diff
@@ -69,15 +77,15 @@ const OdsCollapsible = {
   },
 
   bindAll() {
-    window.addEventListener('OdsBreakpointChange', handleBreakpointChange);
+    window.addEventListener('OsgBreakpointChange', handleBreakpointChange);
```
Comment (larsboldt) [inline src/general/helpers/collapsible/collapsible.js]: ods

File: src/general/helpers/collapsible/collapsible.js [hunk 5]
```diff
@@ -69,15 +77,15 @@ const OdsCollapsible = {
...
     triggerIterator((item) => {
       item.addEventListener('click', handleCollapsible);
       item.addEventListener('keypress', handleCollapsible);
     });
   },
 
   unbindAll() {
-    window.removeEventListener('OdsBreakpointChange', handleBreakpointChange);
+    window.removeEventListener('OsgBreakpointChange', handleBreakpointChange);
```
Comment (larsboldt) [inline src/general/helpers/collapsible/collapsible.js]: ods

File: src/general/helpers/collapsible/collapsible.init.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+
+document.addEventListener('DOMContentLoaded', () => {
+  OsgCollapsible.init();
```
Comment (larsboldt) [inline src/general/helpers/collapsible/collapsible.init.js]: ods

**Review by larsboldt — CHANGES_REQUESTED** [review]

File: src/components/accordion/accordion.json [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{
+  "version": "1.1",
```
[the remark this answers is not in the document]
Comment (cristain7) [reply src/components/accordion/accordion.json:2]: Absolutt 1.0!

**Comment by cristain7** [comment]
> Siden dette er et nytt komponent, er det testet på flere enheter og nettlesere for rariteter?

@kbergha jeg har testet i flere nettlesere og mobil

**Review by kbergha — APPROVED** [review]

File: src/general/helpers/collapsible/collapsible.init.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import OsdCollapsible from './collapsible';
+
+document.addEventListener('DOMContentLoaded', () => {
+  OsdCollapsible.init();
```
Comment (larsboldt) [inline src/general/helpers/collapsible/collapsible.init.js:4]: Osd? Ods...

File: src/general/helpers/collapsible/collapsible.js [hunk 9]
```diff
@@ -7,6 +7,12 @@ function triggerIterator(callback) {
   });
 }
 
+function handleAnimation(e) {
+  if (e.target.classList.contains('ods-animate')) {
```
Comment (larsboldt) [inline src/general/helpers/collapsible/collapsible.js:11]: what is ods-animate and ods-rotate?

File: src/components/accordion/accordion.scss [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+    }
+  }
+
+  & .ods-rotate::after {
+    transform: translate(-50%, -50%) rotate(180deg);
+  }
+
+  &__content {
+    @extend %ods-padding-5, %ods-padding-bottom-8;
+    @extend %ods-padding-top-6-breakpoint-large,  %ods-padding-bottom-8-breakpoint-large, %ods-padding-horizontal-8-breakpoint-large;
```
Comment (larsboldt) [inline src/components/accordion/accordion.scss:105]: remove extra space

**Review by larsboldt — CHANGES_REQUESTED** [review]

File: src/general/helpers/collapsible/collapsible.js (hunk 9 above)
Comment (cristain7) [reply src/general/helpers/collapsible/collapsible.js:11]: ods-animate is a class flag to run animation or not.
ods-rotate is the class of the animation that rotates the arrow.

Comment (larsboldt) [reply src/general/helpers/collapsible/collapsible.js:11]: But why are they not part of the pull request? Is this a new system that we have implemented or is it already in designsystem?

Comment (cristain7) [reply src/general/helpers/collapsible/collapsible.js:11]: They should be! Will check what happened here

Comment (cristain7) [reply src/general/helpers/collapsible/collapsible.js:11]: ods-animate and ods-rotate are part of this review, they are used only in this function handleAnimation(e)

Comment (larsboldt) [reply src/general/helpers/collapsible/collapsible.js:11]: I don't get it - where is the css for ods-rotate defined?

Comment (larsboldt) [reply src/general/helpers/collapsible/collapsible.js:11]: What is the concept behind ods-animate, where is it documented? A generic name like this must refer to a system documented somewhere?

Comment (cristain7) [reply src/general/helpers/collapsible/collapsible.js:11]: Good point! It is part of the collapsible helper and should be documented there.
Also ods-rotate is defined i accordion (wrooooong) that's why you didn't find it. Will move it to helper

File: src/general/helpers/collapsible/collapsible.scss [hunk 11]
```diff
@@ -0,0 +1,8 @@
+.ods-collapsible--rotate::after {
+  transform: translate(-50%, -50%) rotate(180deg);
+}
+
+.ods-collapsible__trigger::after {
```
Comment (larsboldt) [inline src/general/helpers/collapsible/collapsible.scss:5]: Have you checked that this does not affect any current triggers?

Comment (cristain7) [reply src/general/helpers/collapsible/collapsible.scss:5]: It is the only place using this class name

Comment (cristain7) [reply src/general/helpers/collapsible/collapsible.scss:5]: I didn't mean to add this file

Comment (larsboldt) [reply src/general/helpers/collapsible/collapsible.scss:5]: Isn't this related to rotate, should not the classname reflect that?

Comment (cristain7) [reply src/general/helpers/collapsible/collapsible.scss]: The purpose of removing the animation functionality from the accordion component is to create an animation helper. So it shouldn't be on the collapsible helper either.

Comment (cristain7) [reply src/general/helpers/collapsible/collapsible.scss]: I removed this file

**Review by larsboldt — APPROVED** [review]

**Pull request merged**